src/java/picard/sam/DownsamplingIterator.java

```diff
@@ -0,0 +1,125 @@
+package picard.sam;
```

Comment:
I like the idea of iterators being available in htsjdk, so folks do not need to pull in Picard.  This tool is *really* useful as so many folks downsample.